packages/components/src/mobile/link-settings/test/edit.native.js

```diff
@@ -69,44 +69,32 @@ describe.each( [
 	 * GIVEN the CLIPBOARD has a URL copied;
 	 * WHEN the USER selects the SETTINGS BUTTON on the EDIT IMAGE BLOCK or EDIT BUTTON BLOCK;
 	 */
-	// eslint-disable-next-line jest/no-done-callback
-	it( 'should display the LINK SETTINGS with an EMPTY LINK TO field.', async ( done ) => {
+	it( 'should display the LINK SETTINGS with an EMPTY LINK TO field.', async () => {
 		// Arrange
-		const expectation =
-			'The LINK SETTINGS > LINK TO field SHOULD be displayed WITHOUT a URL from the CLIPBOARD.';
 		const url = 'https://tonytahmouchtest.files.wordpress.com';
 		const subject = await initializeEditor( { initialHtml } );
 		Clipboard.getString.mockReturnValue( url );
 
 		// Act
-		try {
-			const block = await waitFor( () =>
-				subject.getByA11yLabel(
-					type === 'core/image' ? /Image Block/ : /Button Block/
-				)
-			);
-			fireEvent.press( block );
-			fireEvent.press( block );
-			fireEvent.press(
-				await waitFor( () => subject.getByA11yLabel( 'Open Settings' ) )
-			);
-		} catch ( error ) {
-			done.fail( error );
-		}
+		const block = await waitFor( () =>
+			subject.getByA11yLabel(
+				type === 'core/image' ? /Image Block/ : /Button Block/
+			)
+		);
+		fireEvent.press( block );
+		fireEvent.press( block );
+		fireEvent.press(
+			await waitFor( () => subject.getByA11yLabel( 'Open Settings' ) )
+		);
 
 		// Assert
-		try {
-			await waitFor( () =>
-				subject.getByA11yLabel(
-					`Link to, ${
-						type === 'core/image' ? 'None' : 'Search or type URL'
-					}`
-				)
-			);
-			done();
-		} catch ( error ) {
-			done.fail( expectation );
-		}
+		await waitFor( () =>
+			subject.getByA11yLabel(
+				`Link to, ${
+					type === 'core/image' ? 'None' : 'Search or type URL'
+				}`
+			)
+		);
```

Comment:
I resolved [this](https://github.com/WordPress/gutenberg/pull/37715/files?diff=split&w=0#r781499147) conversation because the intention was to determine if removing `done` usages is okay (which I think it's fine). 

However, I also mentioned something similar regarding the removal of the expectations from the test implementations. I don't feel super strongly regarding whether we keep them. I just felt that they described what the intent behind the existing UX should be in case the UX requirements changed in the future. I remember at one point trying to combine the error description with the one emitted from `waitFor` as to not lose the original error information. That would involve needing to emit a new error instance and decorate it with the description from the previously caught error though, e.g.,

```
try {
  await waitFor(() => subject.getBy*());
} catch ({message}) {
  const expectation = '';
  throw new Error(`${message}\n\n${expectation}`);
}
```

or something similar.

Reply:
> I'm wondering if we might consider adding an expectation, wdyt?

@fluiddot great question! **TL;DR:** I removed the `expectation`s as I did not see a clear path towards keeping them, but the `throw` approach @ttahmouch provided in his comment looks great. I am happy to reinstate the `expectation`s using `throw`. 👍🏻 

That said, I'll share my own perspective for consideration. I am curious of each of your thoughts. 

![Diagramed Jest test output](https://user-images.githubusercontent.com/438664/148845767-4b108641-0196-430f-b8db-1cc61a6ba875.png)

With Jest — and Jasmine before it — I am accustom to communicating a test's expectation in the test description with `describe` and `it` (yellow) and allowing the actual error message (purple) to show up along side it as well, as that naturally occurs. 

From my understanding of this thread's proposal — and the original code — we would create our own log message (green) to combine the actual error (purple) with an expectation (orange). From my perspective, the approach feels a bit duplicative of the test description, but I am may be overlooking someting. 

Either way, I am happy to reinstate the removed `expectation`s. Thanks! 🙇🏻

Reply:
> I'm wondering if we might consider adding an expectation, wdyt?

@fluiddot rereading your comment, I wonder if now if you were primarily referring to adding assertions like we did in https://github.com/WordPress/gutenberg/pull/37715/commits/e0659215d317502aa20aab797f79ed110f1342a5. Either way, I am still interested in feedback on my my initial response.

Reply:
Very interesting information here, I agree with @dcalhoun in relation to the fact that the expectation message is already included in the test description, so feels a bit redundant to display it a second time when the test fails.

On the other hand, I see a potential improvement in the `waitFor` function for cases where we want to provide extra information along with the error message if the wait times out. In relation to this, we could include a new string parameter that could be appended or prepended to the exception throw:
https://github.com/WordPress/gutenberg/blob/d7c7af335f6108df6db673d5876de976e3fa2c73/test/native/helpers.js#L99

In any case, I'd consider this as an improvement to be tackled in a future PR.

> @fluiddot rereading your comment, I wonder if now if you were primarily referring to adding assertions like we did in e065921. Either way, I am still interested in feedback on my my initial response.

Yep, I was referring to include `expect` statements. I know it wouldn't be actually necessary for the tests, but I'm accustomed to always having at least one expectation in a test, at least semantically to outline the assertion part of it.

Reply:
Thanks for the feedback, @fluiddot. I added explicit assertions where they were absent in https://github.com/WordPress/gutenberg/pull/37715/commits/e703034450513b1ddb789c2c209e08559a566eb2. I will leave this conversation as unresolved for future reference, in case someone would like to explore adding the ability to conditionally log additional information in a separate PR.

Reply:
**My (Irrelevant) Opinion:** When tests are integration-style (tending toward functional-style), and test the surface of the UI components and not the underlying implementation details of the code, you end up with a lot more staging steps to get to the final assertion just given the nature of pretending to interact with the UI as the customer. So the goal would be if one of those leading steps fails, then notify us of the reason it failed relevant to that step. You can still infer the reason for the failure from the element that it may have failed to resolve in a query. However, the goal was just to make it more clear in prose.

I think where I failed in my original test implementations is making the expectation for the steps leading up to the final assertion relevant to the steps themselves. (This may also just be completely unnecessary, and I may just be overcomplicating things that don't add much value. 🤷‍♂️)

I fully trust whatever decision you guys decide is best for the sake of moving forward with fixing the timeouts. This isn't that important or urgent to me. I also appreciate you reaching out about it.

Reply:
It sounds like we can resolve this (and related) threads, and just backlog future enhancements if I understood the full context from this thread. I just didn't want to preemptively resolve it in case you decide to move forward with it here.